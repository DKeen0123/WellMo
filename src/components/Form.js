import React, { Component } from 'react';
import Scale from './Scale';
import Header from './Header';

class Form extends Component {
  constructor() {
    super();
    this.state = { firstScaleScore: undefined };
  }

  updateState = event => this.setState({ firstScaleScore: event.target.value });
  displayScore = () => {
    const outputBox = document.getElementById('outputBox');
    outputBox.innerHTML = this.state.firstScaleScore;
  };

  render() {
    return (
      <div>
        <Header />
        <Scale updateState={this.updateState} />
        <button id="displayButton" onClick={this.displayScore}>
          Enter Mood
        </button>
        <p id="outputBox" />
      </div>
    );
  }
}

export default Form;
