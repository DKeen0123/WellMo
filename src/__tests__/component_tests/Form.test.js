import React from 'react';
import { shallow } from 'enzyme';
import Form from '../../components/Form';

describe('Form', () => {
  let form = shallow(<Form />);

  it('renders correctly', () => {
    expect(form).toMatchSnapshot();
  })
});