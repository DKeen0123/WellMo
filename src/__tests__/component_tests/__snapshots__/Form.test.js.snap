// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
         Hot Thought Form
</h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": " Hot Thought Form",
      },
      "ref": null,
      "rendered": " Hot Thought Form",
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
           Hot Thought Form
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Hot Thought Form",
        },
        "ref": null,
        "rendered": " Hot Thought Form",
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
